cmd/transportRequestUploadCTS.go

```diff
@@ -0,0 +1,106 @@
+package cmd
+
+import (
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/transportrequest"
+)
+
+type transportRequestUploadUtils interface {
+	command.ShellRunner
+	GetAction() CTSUploadAction
+
+	// Add more methods here, or embed additional interfaces, or remove/replace as required.
+	// The transportRequestUploadUtils interface should be descriptive of your runtime dependencies,
+	// i.e. include everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+// ActionProvider ...
+type ActionProvider struct {
+	action CTSUploadAction
+}
+
+// CTSUploadAction ...
+type CTSUploadAction interface {
+	Perform(command.ShellRunner) error
+	WithConnection(transportrequest.CTSConnection)
+	WithApplication(transportrequest.CTSApplication)
+	WithNodeProperties(transportrequest.CTSNode)
+	WithTransportRequestID(string)
+	WithConfigFile(string)
+	WithDeployUser(string)
+}
+
+// GetAction ...
+func (provider *ActionProvider) GetAction() CTSUploadAction {
```

Comment:
why is this exported?

Reply:
there is no real need to export that. --> Adjusted